### PR TITLE
Filter on multiple Access Levels, Authors, Categories, and Tags (continuation)

### PR DESCRIPTION
**Pull request by florian1995**
Pull Request for Issue #5474 .

I try to finish the great work from @jrseliga.
I fixed the problem that when you deselect for example all tags the list won't refresh. With the example code from @ggppdk it work's like a charm. Thanks for that.
I tested it with MySQL and PostgreSQL and it worked fine. It will be great if someone could test it also with MSSQL.

The following test instructions are also from @jrseliga.
### Articles
1. Create 3 articles:
   1. Article 1 - Category 1 - Tag 1 - Author 1 - Public
   2. Article 2 - Category 2 - Tag 1 & Tag 2 - Author 2 - Registered
   3. Article 3 - Category 2 - Tag 2 - Author 1 - Super User
2. View Articles
   1. Use Search Tools

Just with a simple structure like this, there is a large number of potential test cases. Please refer to the New behavior section for an overview of the filtering logic 
### Featured Articles
1. View Featured Articles
   Use Search Tools
### Categories
1. View Categories
   Use Search Tools

Developed@icampus

### CODE REVIEW

**Comment by zero-24** [comment]
Hi @florian1995 i have fixed some Travis issues and a bug see: https://github.com/florian1995/joomla-cms/pull/1 :+1:

File: administrator/components/com_categories/models/categories.php [hunk 1]
```diff
@@ -95,7 +97,17 @@ protected function populateState($ordering = 'a.lft', $direction = 'asc')
...
 		$this->setState('filter.language', $this->getUserStateFromRequest($this->context . '.filter.language', 'filter_language', '', 'string'));
-		$this->setState('filter.tag', $this->getUserStateFromRequest($this->context . '.filter.tag', 'filter_tag', '', 'string'));
+
+		$formSubmited = JFactory::getApplication()->input->post->get('form_submited');
+
+		$tag = $this->getUserStateFromRequest($this->context . '.filter.tag', 'filter_tag', '');
+
+		if ($formSubmited)
+		{
+			$tag = $app->input->post->get('tag')
```
Comment (zero-24) [inline administrator/components/com_categories/models/categories.php]: @florian1995 sorry I have made a mistake can you please add the missing `;` here? Thanks

**Comment by florian1995** [comment]
Thank you really much @zero-24 for the quick response.
Now the code looks definitely better.
I added the missing semicolons and merged your code, now all build checks have passed, thank you :)

**Comment by zero-24** [comment]
:+1:

**Comment by ggppdk** [comment]
The labels for the multi-select fields are shown as placeholders
- but when chosen attaches to the multi-select elements, the width of the input element is not calculated correctly, you need to click to open close the chosen element so that it is calculated correctly

To overcome the initial wrong inline CSS:
```
style="width: 0px;"
```

this CSS can be used as a workaround
```
ul.chosen-choices .search-field > input {
    width: 100% !important;
}
```

any better solution ?

**Comment by ggppdk** [comment]
I forgot to say that wrong width calculation happens only,
- when the filters slider is closed

if you select any filter
then the slider is open on page load and the labels inside multi-select elements are shown correctly

**Comment by florian1995** [comment]
Thanks for your reply @ggppdk.
If i understand you right the problem is, that sometimes the placeholders are not shown correctly, because the width isnt  correct. But i tested it with Chrome and Firefox and dont know how to reproduce the issue, i can select, deselect, open and close the filters how i want, but it never happens. Maybe you can say in detail how to reproduce this issue.

**Comment by ggppdk** [comment]
Only thing i can say that the issue occurs (or can occur) when the selects are initially hidden,
because the slider is closed

There is a relevant issue still open in chosen tracker:
https://github.com/harvesthq/chosen/issues/92

- i am using **select2 JS for my multi-select filters**, and select2 JS in general in all my views, maybe if chosen JS had been updated , i would have been using it,

so i do not know much about it

i think suggested solution was to force it via CSS, 

or add some JS code that .... e.g. will do something on slider open ...

**Comment by roland-d** [comment]
@ggppdk Do you have any exact steps for reproducing the missing placeholder? I have tried to make it disappear but apparently I am not a magician. Thanks.

**Comment by patjun** [comment]
testing<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11517">issues.joomla.org/tracker/joomla-cms/11517</a>.</sub>

**Comment by patjun** [comment]
@ggppdk do you still have the problem with Joomla 3.7.2. I try to reproduce the problem, but i cant.
Please answer in the next days otherwise we will delete the issue. @nibra <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11517">issues.joomla.org/tracker/joomla-cms/11517</a>.</sub>

**Comment by ggppdk** [comment]
I have not tested with j3.7.2 i guess if you tested and it now works , then it has been fixed

**Comment by Schmidie64** [comment]
I have tested this item :white_check_mark: successfully on 8a95b4aa30e26f3e51783c1e7c2688e1555834b3<br /><br />@Icampus
I've tested this Issue succesfull. Everything works fine. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11517">issues.joomla.org/tracker/joomla-cms/11517</a>.</sub>

**Comment by DrDreave** [comment]
I have tested this item :white_check_mark: successfully on 8a95b4aa30e26f3e51783c1e7c2688e1555834b3<br /><br />

# Operating System
* Joomla! 3.8.0-beta3-dev
* PHP 5.6.2
* MySQLi 5.5.38
* Apache/2.2.29 (Unix)

# Steps
* Open article administration
* Use search tools

# Test before patch
* Single filter selection
Single filter selection

# Test after patch
* Multi filter selection for category, access, author tag
Multi filter selection

Tested @icampus<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11517">issues.joomla.org/tracker/joomla-cms/11517</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests. 

Please resolve conflicting Files.

**Comment by wojsmol** [comment]
@franz-wohlkoenig This PR has conflicts - see on github.com.

**Comment by THMmeddah** [comment]
I have tested this item :white_check_mark: successfully on 8a95b4aa30e26f3e51783c1e7c2688e1555834b3<br /><br />tested successfully<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11517">issues.joomla.org/tracker/joomla-cms/11517</a>.</sub>

**Comment by THMmeddah** [comment]
testing<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11517">issues.joomla.org/tracker/joomla-cms/11517</a>.</sub>

**Comment by ghost** [comment]
@THMmeddah thanks for Test. We need 2 successfully Test for an Pull Request.

**Comment by AresRoxx10** [comment]
I have tested this item :white_check_mark: successfully on 8a95b4aa30e26f3e51783c1e7c2688e1555834b3<br /><br />Tested successfully<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11517">issues.joomla.org/tracker/joomla-cms/11517</a>.</sub>

**Comment by ghost** [comment]
@AresRoxx10 thanks for Test. We need 2 successfully Test for an Pull Request, we have now 4 :-).

**Comment by roland-d** [comment]
@franz-wohlkoenig Issue can be closed as implemented in #17668

**Comment by brianteeman** [comment]
Closed as requested

**Pull request closed without merging**